homeassistant/components/tplink/sensor.py

```diff
@@ -73,63 +76,50 @@ class TPLinkSensorEntityDescription(SensorEntityDescription):
         native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
         device_class=SensorDeviceClass.ENERGY,
         state_class=SensorStateClass.TOTAL_INCREASING,
+        feature_id="consumption_today",
         precision=3,
     ),
     TPLinkSensorEntityDescription(
         key=ATTR_VOLTAGE,
         native_unit_of_measurement=UnitOfElectricPotential.VOLT,
         device_class=SensorDeviceClass.VOLTAGE,
         state_class=SensorStateClass.MEASUREMENT,
-        emeter_attr="voltage",
+        feature_id="voltage",
         precision=1,
     ),
     TPLinkSensorEntityDescription(
         key=ATTR_CURRENT_A,
         native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
         device_class=SensorDeviceClass.CURRENT,
         state_class=SensorStateClass.MEASUREMENT,
-        emeter_attr="current",
+        feature_id="current",
         precision=2,
     ),
 )
 
 
-def async_emeter_from_device(
-    device: Device, description: TPLinkSensorEntityDescription
-) -> float | None:
-    """Map a sensor key to the device attribute."""
-    if attr := description.emeter_attr:
-        if (val := getattr(device.emeter_realtime, attr)) is None:
-            return None
-        return round(cast(float, val), description.precision)
-
-    # ATTR_TODAY_ENERGY_KWH
-    if (emeter_today := device.emeter_today) is not None:
-        return round(cast(float, emeter_today), description.precision)
-    # today's consumption not available, when device was off all the day
-    # bulb's do not report this information, so filter it out
-    return None if device.device_type == Device.Type.Bulb else 0.0
-
-
-def _async_sensors_for_device(
-    device: Device,
-    coordinator: TPLinkDataUpdateCoordinator,
-    parent: Device | None = None,
-) -> list[CoordinatedTPLinkEntity]:
-    """Generate the sensors for the device."""
-    sensors: list[CoordinatedTPLinkEntity] = []
-    if device.has_emeter:
-        sensors = [
-            SmartPlugSensor(device, coordinator, description, parent=parent)
-            for description in ENERGY_SENSORS
-            if async_emeter_from_device(device, description) is not None
-        ]
-    new_sensors: list[CoordinatedTPLinkEntity] = [
-        Sensor(device, coordinator, feat, parent=parent)
-        for feat in device.features.values()
-        if feat.type == Feature.Sensor
-    ]
-    return sensors + new_sensors
+def _new_sensor_description(feature: Feature) -> TPLinkSensorEntityDescription:
+    """Create description for entities not yet added to the static description list."""
```

Comment:
I wonder if the `new` entities should be disabled by default?

Reply:
I don't think so, we should have a single straightforward way to handle this to keep the "integration shim" as simple as possible. If we are really going to the direction where we use descriptions to describe our features, we should just log a warning informing that the integration has encountered a feature that doesn't have a definition.